src/components/Menu/MenuItem/AvatarMenuItem.tsx

```diff
@@ -0,0 +1,27 @@
+import React, { forwardRef, useCallback } from "react";
+import MenuItem from "./MenuItem";
+import Avatar, { AvatarProps } from "../../Avatar/Avatar";
+import VibeComponentProps from "../../../types/VibeComponentProps";
+
+interface AvatarMenuItemProps extends VibeComponentProps {
+  avatarProps?: AvatarProps;
+  // TODO ts-migration replace with MenuItemProps when MenuItem is converted to TS
+  menuItemProps?: any;
+}
+
+/**
+ * MenuItem with Avatar instead of Icon
+ */
+const AvatarMenuItem: React.FC<AvatarMenuItemProps> = forwardRef(({ avatarProps, ...menuItemProps }, ref) => {
+  const renderAvatar = useCallback(() => <Avatar {...avatarProps} />, [avatarProps]);
+
+  // @ts-ignore TODO ts-migration: remove this line & fix the issues when MenuItem is converted to TS
+  return <MenuItem {...menuItemProps} icon={renderAvatar} ref={ref} />;
+});
+
+// @ts-ignore TODO how to solve this?
+AvatarMenuItem.isMenuChild = true;
```

Comment:
Add isMenuChild and isSelectable to the component interface by using & as we did for the enums in the loader component.
Also let's change the way we assign values to additional attributes on the component to using object.assign for consistency

Reply:
No questions about Object.assign, but are you sure we need to add it to component interface? If we do so they are just seen in component's props autocomplete and I'm not sure that the user really need to know about them - they are kinda inner